layer/mounted_layer.go

```diff
@@ -146,6 +156,13 @@ func (rl *referencedRWLayer) Unmount() error {
 		return ErrLayerNotRetained
 	}
 	rl.activityCount--
+	if rl.activityCount > 0 {
+		return ErrActiveMount
```

Comment:
Likewise, do you think we really need to return `ErrNotMounted` when it's already 0?
We shouldn't get to that point, but...